src/address.rs

```diff
@@ -94,19 +95,21 @@ impl fmt::String for Mailbox {
 }
 
 impl FromStr for Mailbox {
-    fn from_str(s: &str) -> Option<Mailbox> {
+    type Err = ParsingError;
+
+    fn from_str(s: &str) -> ParsingResult<Mailbox> {
         AddressParser::new(s).parse_mailbox()
     }
 }
 
 impl FromHeader for Vec<Address> {
-    fn from_header(value: String) -> Option<Vec<Address>> {
-        Some(AddressParser::new(value.as_slice()).parse_address_list())
+    fn from_header(value: String) -> ParsingResult<Vec<Address>> {
+        AddressParser::new(value.as_slice()).parse_address_list()
     }
 }
 
 impl ToFoldedHeader for Vec<Address> {
-    fn to_folded_header(start_pos: usize, value: Vec<Address>) -> Option<String> {
+    fn to_folded_header(start_pos: usize, value: Vec<Address>) -> ParsingResult<String> {
```

Comment:
Returning a ``ParsingResult`` in APIs that don't actually parse anything is a bit ugly. Any better naming suggestions, or should we tell people to deal w/ it?